### PR TITLE
Add a test for 'content-to-slot-usages' fixes for paper-icon-item.

### PR DESCRIPTION
**Pull request by bicknellr**
- ~~Adds `slot="item-icon"` to children of `<paper-icon-item>` that don't already have a slot, but have an `item-icon` attribute.~~
- [x] #118 has been merged.
 - [x] CHANGELOG.md has been updated

---
This change is [Reviewable](https://reviewable.io/reviews/polymer/polymer-linter/119)

### CODE REVIEW

**Comment by bicknellr** [comment]
Needs tests.

File: src/polymer/elements/paper-item-v1-to-v2.ts [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+import {registry} from '../../registry';
+import {HtmlRule} from '../../html/rule';
+import {ParsedHtmlDocument, Severity, Warning} from 'polymer-analyzer';
+import {FixableWarning} from '../../warning';
+
+class PaperItemV1ToV2 extends HtmlRule {
+  code = 'paper-item-v1-to-v2';
+  description = stripIndent(`
+    ADD A DESCRIPTION
```
Comment (rictic) [inline src/polymer/elements/paper-item-v1-to-v2.ts]: ^

Comment (bicknellr) [inline src/polymer/elements/paper-item-v1-to-v2.ts]: Needs a description.

File: src/polymer/elements/paper-item-v1-to-v2.ts [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+      if (node.tagName !== undefined &&
+        !dom5.hasAttribute(node, 'slot') &&
+        dom5.hasAttribute(node, 'item-icon')
+      ) {
+        const startTagSourceRange =
+          parsedDocument.sourceRangeForStartTag(node)!;
+        const warning = new FixableWarning({
+          parsedDocument,
+          code: this.code,
+          message: 'ADD MESSAGE',
```
Comment (bicknellr) [inline src/polymer/elements/paper-item-v1-to-v2.ts]: Needs a message.

Comment (rictic) [inline src/polymer/elements/paper-item-v1-to-v2.ts]: ^

File: src/polymer/elements/paper-item-v1-to-v2.ts [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+      let suspectNodes = Array.from(paperIconItem.childNodes!);
+
+      const nodeIsTemplateExtension = (node: dom5.Node) => {
+        if (!node.attrs) return false;
+
+        const isAttr = node.attrs.find(attr => attr.name === 'is');
+        return (
+          node.tagName === 'template' &&
+          isAttr &&
+          ['dom-bind', 'dom-if', 'dom-repeat'].includes(isAttr.value)
```
Comment (bicknellr) [inline src/polymer/elements/paper-item-v1-to-v2.ts]: Is this the complete list?

File: src/polymer/elements/paper-item-v1-to-v2.ts [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+        }];
+
+        warnings.push(warning);
+      }
+    };
+
+    for (const paperIconItem of paperIconItems) {
+      let suspectNodes = Array.from(paperIconItem.childNodes!);
+
+      const nodeIsTemplateExtension = (node: dom5.Node) => {
```
Comment (rictic) [inline src/polymer/elements/paper-item-v1-to-v2.ts]: Consider pulling this function out into `src/html/util.ts`

File: src/polymer/elements/paper-item-v1-to-v2.ts [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+
+import * as dom5 from 'dom5';
+import stripIndent = require('strip-indent');
+
+import {registry} from '../../registry';
+import {HtmlRule} from '../../html/rule';
+import {ParsedHtmlDocument, Severity, Warning} from 'polymer-analyzer';
+import {FixableWarning} from '../../warning';
+
+class PaperItemV1ToV2 extends HtmlRule {
```
Comment (rictic) [inline src/polymer/elements/paper-item-v1-to-v2.ts]: Can this be just a static predicate in `content-to-slot-usages`?

File: src/polymer/elements/paper-item-v1-to-v2.ts (hunk 4 above)
Comment (bicknellr) [reply src/polymer/elements/paper-item-v1-to-v2.ts]: I'll do this in #118 and then use it here.

File: src/polymer/elements/paper-item-v1-to-v2.ts (hunk 3 above)
Comment (bicknellr) [reply src/polymer/elements/paper-item-v1-to-v2.ts]: Fixed in 35c075d.

File: src/polymer/elements/paper-item-v1-to-v2.ts (hunk 1 above)
Comment (bicknellr) [reply src/polymer/elements/paper-item-v1-to-v2.ts]: fixed

File: src/polymer/elements/paper-item-v1-to-v2.ts (hunk 2 above)
Comment (bicknellr) [reply src/polymer/elements/paper-item-v1-to-v2.ts]: fixed

File: src/polymer/elements/paper-item-v1-to-v2.ts (hunk 5 above)
Comment (bicknellr) [reply src/polymer/elements/paper-item-v1-to-v2.ts]: Wow, yeah, I should just do this...

Comment (bicknellr) [reply src/polymer/elements/paper-item-v1-to-v2.ts]: Hmm, it looks like it's already there, maybe I can drop all of the non-test content of this PR?

File: src/polymer/elements/paper-item-v1-to-v2.ts [hunk 6]
```diff
@@ -0,0 +1,111 @@
+/**
```
Comment (rictic) [inline src/polymer/elements/paper-item-v1-to-v2.ts]: Should removing the hardcoded paper-item rule in content-to-slot-usages

Comment (rictic) [inline src/polymer/elements/paper-item-v1-to-v2.ts]: This makes sense as part of the Polymer 2 and Polymer 2 Hybrid collections

Comment (rictic) [inline src/polymer/elements/paper-item-v1-to-v2.ts]: Please update the CHANGELOG

**Review by rictic — COMMENTED** [review]
These changes are nice. They're a step up in detail and specificity from the other content-to-slot changes, so I think it's fine to merge them.

**Comment by bicknellr** [comment]
Sorry, made this last commit before seeing your comments. Should I revert back to the separate rule?

**Comment by rictic** [comment]
Eh, either is good. Up to you whichever one you'd prefer

**Comment by bicknellr** [comment]
Given that 'content-to-slot-usages' seems to already cover this situation, keeping it there probably makes more sense?

**Review by rictic — APPROVED** [review]
+1

**Pull request merged**